Tests fail on Alpine Linux with git 2.27.0

Hi,

Trying to get vcstool onto Alpine linux and the tests seem to be failing. I had to run this command first before they would pass `git config --global pull.rebase false`.

```
/package $ git --version
git version 2.27.0
```

```
______________________________________________________________________________________________ TestCommands.test_pull _______________________________________________________________________________________________

self = <test_commands.TestCommands testMethod=test_pull>

    def test_pull(self):
        with self.assertRaises(subprocess.CalledProcessError) as e:
            run_command('pull', args=['--workers', '1'])
        expected = get_expected_output('pull')
        # replace message from older git versions
        output = e.exception.output.replace(
            b'anch. Please specify which\nbranch you want to merge with. See',
            b'anch.\nPlease specify which branch you want to merge with.\nSee')
>       self.assertEqual(output, expected)
E       AssertionError: b'EE.[30 chars]===\nwarning: Pulling without specifying how t[2764 chars]e.\n' != b'EE.[30 chars]===\nYou are not currently on a branch.\nPleas[392 chars]e.\n'

test/test_commands.py:99: AssertionError
____________________________________________________________________________________________ TestCommands.test_pull_api _____________________________________________________________________________________________

self = <test_commands.TestCommands testMethod=test_pull_api>

    def test_pull_api(self):
        try:
            from cStringIO import StringIO
        except ImportError:
            from io import StringIO
        from vcstool.commands.pull import main
        stdout_stderr = StringIO()
    
        # change and restore cwd
        cwd_bck = os.getcwd()
        os.chdir(TEST_WORKSPACE)
        try:
            # change and restore USE_COLOR flag
            from vcstool import executor
            use_color_bck = executor.USE_COLOR
            executor.USE_COLOR = False
            try:
                # change and restore os.environ
                env_bck = os.environ
                os.environ = dict(os.environ)
                os.environ.update(
                    LANG='en_US.UTF-8',
                    PYTHONPATH=(
                        os.path.dirname(os.path.dirname(__file__)) +
                        os.pathsep + os.environ.get('PYTHONPATH', '')))
                try:
                    rc = main(
                        args=['--workers', '1'],
                        stdout=stdout_stderr, stderr=stdout_stderr)
                finally:
                    os.environ = env_bck
            finally:
                executor.USE_COLOR = use_color_bck
        finally:
            os.chdir(cwd_bck)
    
        assert rc == 1
        # replace message from older git versions
        output = stdout_stderr.getvalue().replace(
            'anch. Please specify which\nbranch you want to merge with. See',
            'anch.\nPlease specify which branch you want to merge with.\nSee')
        expected = get_expected_output('pull').decode()
>       assert output == expected
E       AssertionError: assert 'EE..\n=== ./...up to date.\n' == 'EE..\n=== ./...up to date.\n'
E           EE..
E           === ./immutable/hash (git) ===
E         + warning: Pulling without specifying how to reconcile divergent branches is
E         + discouraged. You can squelch this message by running one of the following
E         + commands sometime before your next pull:
E         + 
E         +   git config pull.rebase false  # merge (the default strategy)...
E         
E         ...Full output truncated (63 lines hidden), use '-vv' to show

test/test_commands.py:143: AssertionError
============================================================================================== short test summary info ==============================================================================================
FAILED test/test_commands.py::TestCommands::test_pull - AssertionError: b'EE.[30 chars]===\nwarning: Pulling without specifying how t[2764 chars]e.\n' != b'EE.[30 chars]===\nYou are not currently on a branch.\n...
FAILED test/test_commands.py::TestCommands::test_pull_api - AssertionError: assert 'EE..\n=== ./...up to date.\n' == 'EE..\n=== ./...up to date.\n'
=========================================================================================== 2 failed, 15 passed in 39.03s ===========================================================================================
>>> ERROR: vcstool: check failed
```

--- a/test/test_commands.py
+++ b/test/test_commands.py
@@ -96,6 +96,23 @@ class TestCommands(unittest.TestCase):
         output = e.exception.output.replace(
             b'anch. Please specify which\nbranch you want to merge with. See',
             b'anch.\nPlease specify which branch you want to merge with.\nSee')
+        # newer git versions warn on pull with default config
+        if _get_git_version() >= [2, 27, 0]:
+            pull_warning = b"""
+warning: Pulling without specifying how to reconcile divergent branches is
+discouraged. You can squelch this message by running one of the following
+commands sometime before your next pull:
+
+  git config pull.rebase false  # merge (the default strategy)
+  git config pull.rebase true   # rebase
+  git config pull.ff only       # fast-forward only
+
+You can replace "git config" with "git config --global" to set a default
+preference for all repositories. You can also pass --rebase, --no-rebase,
+or --ff-only on the command line to override the configured default per
+invocation.
+"""
+            output = output.replace(pull_warning, b'')
         self.assertEqual(output, expected)
 
     def test_pull_api(self):
@@ -139,6 +156,23 @@ class TestCommands(unittest.TestCase):
         output = stdout_stderr.getvalue().replace(
             'anch. Please specify which\nbranch you want to merge with. See',
             'anch.\nPlease specify which branch you want to merge with.\nSee')
+        # newer git versions warn on pull with default config
+        if _get_git_version() >= [2, 27, 0]:
+            pull_warning = """
+warning: Pulling without specifying how to reconcile divergent branches is
+discouraged. You can squelch this message by running one of the following
+commands sometime before your next pull:
+
+  git config pull.rebase false  # merge (the default strategy)
+  git config pull.rebase true   # rebase
+  git config pull.ff only       # fast-forward only
+
+You can replace "git config" with "git config --global" to set a default
+preference for all repositories. You can also pass --rebase, --no-rebase,
+or --ff-only on the command line to override the configured default per
+invocation.
+"""
+            output = output.replace(pull_warning, '')
         expected = get_expected_output('pull').decode()
         assert output == expected
 

